langlib/jballerina.java/src/main/ballerina/annotations.bal

```diff
@@ -126,3 +126,13 @@ public const annotation FieldData FieldSet on source external;
 # }
 # ```
 public const annotation ObjectData Binding on class;
+
+# Describes a type, class or function that is a dependency of java native code. This annotation should be added by the
+# developer if a type definition is being accessed by a `io.ballerina.runtime.api.creators` api such as `ValueCreator`.
+# ```ballerina
+# @ExternalDependency
+# function foo(int x) returns int {
+#     return x * 2;
+# }
+# ```
+public const annotation ExternalDependency on source type, source class, source function;
```

Comment:
A better name "dependsOnExternal"?

Reply:
Had an offline discussion with @hasithaa and we decided to use "ExternalDependency" as the annotation name.